internal/build/phase_config_provider.go

```diff
@@ -122,14 +123,34 @@ func WithBinds(binds ...string) PhaseConfigProviderOperation {
 	}
 }
 
-func WithDaemonAccess() PhaseConfigProviderOperation {
+func WithDaemonAccess(dockerHost string) PhaseConfigProviderOperation {
 	return func(provider *PhaseConfigProvider) {
 		WithRoot()(provider)
-		bind := "/var/run/docker.sock:/var/run/docker.sock"
-		if provider.os == "windows" {
-			bind = `\\.\pipe\docker_engine:\\.\pipe\docker_engine`
+		if dockerHost == "inherit" {
+			dockerHost = os.Getenv("DOCKER_HOST")
+		}
+		var bind string
+		if dockerHost == "" {
+			bind = "/var/run/docker.sock:/var/run/docker.sock"
+			if provider.os == "windows" {
+				bind = `\\.\pipe\docker_engine:\\.\pipe\docker_engine`
+			}
+		} else {
+			switch {
+			case strings.HasPrefix(dockerHost, "tcp://"):
+				provider.ctrConf.Env = append(provider.ctrConf.Env, fmt.Sprintf(`DOCKER_HOST=%s`, dockerHost))
+			case strings.HasPrefix(dockerHost, "unix://"):
+				bind = fmt.Sprintf("%s:/var/run/docker.sock", strings.TrimPrefix(dockerHost, "unix://"))
+			case strings.HasPrefix(dockerHost, "npipe://") || strings.HasPrefix(dockerHost, `npipe:\\`):
+				sub := ([]rune(dockerHost))[8:]
+				bind = fmt.Sprintf(`%s:\\.\pipe\docker_engine`, string(sub))
+			default:
+				panic(fmt.Sprintf("don't know how to handle docker-host value: %s", dockerHost))
```

Comment:
I don't think we should panic here, it's not the most graceful exit. How about we just have it be a noop, and write to the errorWriter that we are ignoring the request?

Reply:
It is validate on CLI flag level that we won't get here under normal circumstances. Is that OK?

Reply:
I changed behaviour. For any protocol that is not `unix://` nor `npipe://` the dockerHost flag is simply passed to build container via envvar. So the method never panics.